packages/algob/src/internal/deployer.ts

```diff
@@ -573,6 +574,47 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
 		persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
 	}
 
+	/**
+	 * Save new ASA information(get from txConfirmation) to checkpoint
+	 * @param asaName name of asa
+	 * @param txConfirmation txn confirmation
+	 */
+	addCheckpointASA(asaName: string, txConfirmation: ConfirmedTxInfo): void {
+		const txn = txConfirmation.txn.txn;
+
+		this.cpData.registerASA(this.networkName, asaName, {
+			assetIndex: txConfirmation["asset-index"],
+			creator: algosdk.encodeAddress(txn.apar?.m as Uint8Array),
+			confirmedRound: txConfirmation["confirmed-round"],
+			deleted: false,
+			assetDef: {
+				total: txn.apar?.t as number,
+				decimals: txn.apar?.dc as number,
```

Comment:
Do we need `as number`? If we're using optional chaining then perhaps the result could be `undefined`. Shouldn't we handle that? Otherwise please add a comment

Reply:
I will check it.